server/data/data-api/src/main/java/org/apache/james/user/api/UsersRepository.java

```diff
@@ -142,4 +142,10 @@
      */
     boolean isAdministrator(String username) throws UsersRepositoryException;
 
+    /**
+     * @return true if one can use {@link UsersRepository#updateUser(User)} {@link UsersRepository#addUser(String, String)}
```

Comment:
I still don't buy the complexity argument. Calling things by there names and having smaller interfaces is not complexity.
We agree on the cost part by the way.